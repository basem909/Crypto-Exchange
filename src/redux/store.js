import {
  legacy_createStore as legacyCreateStore,
  combineReducers,
  applyMiddleware,
} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import showCoins from './reducer';
import { showCoinList } from './action-creators';

const allReducers = combineReducers({
  showCoins,
});
const store = legacyCreateStore(allReducers, applyMiddleware(logger, thunk));
store.dispatch(showCoinList());

export default store;
