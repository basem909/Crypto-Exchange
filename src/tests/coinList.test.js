import React from 'react';
import { render, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import CoinsExihbit from "../components/coinList";
import store from '../redux/store';
import "@testing-library/jest-dom/extend-expect";

describe("CoinsExhibit snapshot", () => {
  afterEach(cleanup);

  it("should render CoinsExhibit correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CoinsExihbit />
      </Provider>
    );
    expect(asFragment(<CoinsExihbit />)).toMatchSnapshot();
  });
});

