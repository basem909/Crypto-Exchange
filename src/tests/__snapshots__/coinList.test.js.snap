// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinsExhibit snapshot should render CoinsExhibit correctly 1`] = `
<DocumentFragment>
  <div>
    <header
      class="header"
    >
      <h1
        class="title"
      >
        Crypto World
      </h1>
    </header>
    <img
      alt="crypto"
      class="background"
      src="B.jpeg"
    />
    <form
      class="search-bar"
    >
      <span
        name="search"
      >
         Search by Coin here 
      </span>
      <input
        id="searchBar"
        name="search"
        placeholder="Coin name or symbol"
        type="text"
      />
    </form>
    <div
      class="coins-container"
    />
  </div>
</DocumentFragment>
`;
